index.js

```diff
@@ -125,7 +125,10 @@ Discovery.prototype.destroy = function (cb) {
     self.tracker.removeListener('peer', self._onTrackerPeer)
     self.tracker.removeListener('update', self._onTrackerAnnounce)
     tasks.push(function (cb) {
-      self.tracker.destroy(cb)
+      process.nextTick(function () {
+        self.tracker.destroy(cb)
```

Comment:
This doesn't have the nice 'instant gratification' setup as you see but what I did to get EVENT 3 to send was:
1) https://github.com/feross/bittorrent-tracker/pull/193
2) Completely remove the self.tracker.destroy(cb) and don't add the process.nextTick.
At line 122 you already submit a 'stopped' request and the UDP Tracker client recognizes this and sets up a quick cleanup after it's done. I'll submit my own PR in a minute.

NOTE: I would keep 'self.tracker = null' for JS garbage collection.